### PR TITLE
Update footer to match new mockups

### LINKED ISSUE #2012: Update footer layout
## Footer screenshot
Please see the design issue #2009

The footer layout should be updated to match the screenshot.

## Mockup 
- Footer component: [Design Library, upcoming footer page](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=3018%3A7629&t=gjE9wqTv3hxc2Kek-0).

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #2012 by @obulat
Related to #2009

## Description

This PR updates the footer to match the latest mockups.

The footer layout should be updated to match the screenshot.

## Mockup 
- Footer component: [Design Library, upcoming footer page](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=3018%3A7629&t=gjE9wqTv3hxc2Kek-0).

## Testing Instructions

The footer should work properly, and match the mockups.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
@obulat should there be accompanying updates to visual regression tests here? Thanks for the speedy implementation!

File: src/components/VFooter/VFooter.vue [hunk 1]
```diff
@@ -113,52 +115,91 @@ export default defineComponent({
...
+.footer-xs .logo-and-links {
+  @apply flex-col;
+}
+.footer-xs .language {
+  width: 100% !important;
+}
+.footer-sm .logo-and-links {
+  display: grid;
+  grid-auto-flow: column;
+  grid-template-columns: repeat(2, 1fr);
```
Comment (dhruvkb) [inline src/components/VFooter/VFooter.vue:149]: IMO these CSS rules should be replaced `@apply`-ied Tailwind utilities.

**Review by dhruvkb — CHANGES_REQUESTED** [review]

File: src/components/VFooter/VFooter.vue (hunk 1 above)
Comment (obulat) [inline src/components/VFooter/VFooter.vue:149]:
```suggestion
  @apply grid grid-flow-col grid-cols-1;
```

File: src/components/VFooter/VFooter.vue [hunk 2]
```diff
@@ -99,66 +101,99 @@ export default defineComponent({
...
-    const isContentMode = computed(() => {
-      if (props.mode) {
-        return props.mode === 'content'
-      } else {
-        return ['search', 'audio', 'image'].some((prefix) =>
-          currentPage.value?.startsWith(prefix)
-        )
-      }
-    })
+    const isContentMode = computed(() => props.mode === 'content')
```
Comment (obulat) [inline src/components/VFooter/VFooter.vue:104]: The wrapper component should know whether the footer is in 'content' mode or in the 'internal' mode.

File: src/components/VFooter/VFooter.vue (hunk 1 above)
Comment (obulat) [reply src/components/VFooter/VFooter.vue]: Done :)

**Review by panchovm — CHANGES_REQUESTED** [review]
It looks great, I just noticed the following details.

## Content version
### Spacings
* In `xl` and `2xl` the left and right padding should be `40px` instead of `24px`
* In `lg`, `xl`, and `2xl`, the gap between rows should be `32px` instead of `40px`

### Components
For the page links, I think you used the "Clear" component from the drawer. In theory, that is correct, but the element has a fixed height to match the spacing rule of 8x.

CleanShot 2022-12-01 at 14 26 24@2x

The element has a decimal size of `32.9`, which can be problematic and break layouts if we keep using it as it is. We can tackle this in two ways I think:
1. Replace the Clear component with the [Link component](https://www.figma.com/file/GIIQ4sDbaToCfFQyKMvzr8/Openverse-Design-Library?node-id=3667%3A9753&t=LBXemTz9meYwiYiy-4) and apply the spacing rule to the gap between `li` elements
2. Set a fixed height of the Clear component as `32px` and add a new spacing value between `li` to match the mockup.

I am drawn to the first option as the **Link** component was thought for links single links, whereas the **Clear** component inherits the fixed height from the Results button. Why did this happen? Because the link component did not exist at the moment I started the header redesign.

In `sm`, the locale component has a `max-width: 100%` that crashes the layout. The mockup keeps the width fixed as `230px` in `sm, md, lg, xl, 2xl` to avoid this effect. Is that possible or better to find a different solution?

CleanShot 2022-12-01 at 14 38 33@2x

### Alignment
In `sm` and `md`, the locale component and WordPress call need to be aligned vertically centered.

CleanShot 2022-12-01 at 14 35 32@2x

## Internal version
### Spacings
* In `xl` and `2xl` the left and right padding should be `40px` instead of `24px`
* In `sm` and `md`, padding values should be `24px, 24px, 40px, 24px`. The top part is bigger to match the result area layout.

### Alignment
* In `sm` and `md`, the WordPress call should be fixed on the right side.
* In `sm` and `md`, the locale component and WordPress call need to be aligned vertically centered.

**Review by panchovm — CHANGES_REQUESTED** [review]
It looks great. Just a few notes

## Content version
* Top and bottom paddings must be `40px` in all breakpoints.

## Internal version
* In `lg, xl, 2xl` all sides should be `40px`. I see a `24px` for the bottom side.
* In `sm, md` the top and bottom spacings are inverted. It should be `40px` for the bottom and `24px` for the top.

## Nav items
Despite the above changes. Which one do you think is the best option to address the nav items' components?

**Comment by obulat** [comment]
Thank you for the review, @panchovm! I've fixed the footer paddings.

> Despite the above changes. Which one do you think is the best option to address the nav items' components?

> The element has a decimal size of 32.9, which can be problematic and break layouts if we keep using it as it is.

What kind of problems will the decimal size of 32.9 bring? I don't think we can set a fixed height in the navigation bar because we need the designs to be used in other languages, too. Some navigation items can have a long title when translated, and then they will not fit into the component and will overflow. 

So, we have to set the height using text height + padding. The text height, when it's in one line, is decimal, 16.5px. We either need to set a decimal padding to get a nice link height of 32px, or we have to live with a decimal height of the link:
Screenshot 2022-12-07 at 9 11 11 AM

In the code, we don't really have a `Clear` component, do we have it in the Figma mockups? 
I have added a `VNavLink` component for navigation links. It has an 8px vertical padding to ensure that the links are tappable. I also added some row and column gaps for the footer.

This is what the height becomes for one-line items:
Screenshot 2022-12-07 at 9 17 04 AM

And here, what items with more than one-line of text will look like:
Screenshot 2022-12-07 at 9 40 10 AM

Is this approach OK, or should we think of something else, @panchovm?

**Comment by panchovm** [comment]
> I don't think we can set a fixed height in the navigation bar because we need the designs to be used in other languages

Very much agree with this. My point was about the need for `8px` of padding as the focus style looks strange.

CleanShot 2022-12-07 at 19 16 29@2x

Your point about the tappable area makes sense, but I think we don't need it as the [Target Size Criterion](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html) requires _at least_ a `44px` area.

What you did is definitely correct. I am just bringing this up to reach consistency across other similar elements. I am thinking of the "Clear" action in filters, "Report this content" in the single result, these nav links in the Footer, and possibly others that I am missing.

The `Clear` component does exist in Figma, but since it does not in code, and we are discussing its purpose, it seems timely to define a basic link element that does not have paddings/margins, could have an icon, and where focus ring wraps the element. Something like the image below, but called `VSingleLink` instead (name TBD).

CleanShot 2022-12-07 at 19 24 54@2x

What do you think?

**Comment by obulat** [comment]
> Your point about the tappable area makes sense, but I think we don't need it as the [Target Size Criterion](https://www.w3.org/WAI/WCAG21/Understanding/target-size.html) requires _at least_ a `44px` area.

Today I learned: it's ok for the tappable area (or the target size on the desktop) to be smaller than 44px if the areas do not overlap. So, it's OK from the accessibility point of view (or SEO in lighthouse checks) if the focus area of the links is small, like in the snapshots you shared, @panchovm, but there is enough space between the links. The problem starts when the links are so close to each other that a click on one can be interpreted as a click on another one. And we don't have such problem anyways :) 

I've updated the code to have a small focus "box" around the links. The padding now is on the `<li>` element.

**Review by panchovm — APPROVED** [review]
It looks great. Nice work ✨

File: src/components/VHeader/VWordPressLink.vue [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    class="hover:no-underline focus-visible:rounded-sm focus-visible:outline-none focus-visible:ring focus-visible:ring-pink focus-visible:ring-offset-1 focus-visible:ring-offset-tx"
+  >
+    <i18n
+      tag="p"
+      path="footer.wordpress-affiliation"
+      class="label-regular flex flex-row items-center gap-2"
+    >
+      <template #wordpress>
+        <WordPress class="aria-hidden" />
+        <span class="sr-only">WordPress</span>
```
Comment (zackkrida) [inline src/components/VHeader/VWordPressLink.vue:14]: Great inclusion for a11y.

**Review by zackkrida — APPROVED** [review]
Looks good @obulat!

**Pull request merged**